langchain/src/memory/chat_memory.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  HumanChatMessage,
+  AIChatMessage,
+  BaseChatMessage,
+} from "../schema/index.js";
+import {
+  BaseMemory,
+  InputValues,
+  OutputValues,
+  getInputValue,
+} from "./base.js";
+
+export class ChatMessageHistory {
+  messages: BaseChatMessage[] = [];
+
+  constructor(messages?: BaseChatMessage[]) {
+    this.messages = messages ?? [];
+  }
+
+  addUserMessage(message: string): void {
+    this.messages.push(new HumanChatMessage(message));
+  }
+
+  addAIChatMessage(message: string): void {
+    this.messages.push(new AIChatMessage(message));
+  }
+}
+
+export abstract class ChatMemoryMixin extends BaseMemory {
+  chatHistory: ChatMessageHistory;
+
+  constructor(chatHistory?: ChatMessageHistory) {
+    super();
+    this.chatHistory = chatHistory ?? new ChatMessageHistory();
+  }
+
+  async saveContext(
+    inputValues: InputValues,
+    OutputValues: Promise<OutputValues>
```

Comment:
This is a fairly strange API, receiving a promise of the argument. Should we take this opportunity to fix it?

Reply:
yeah this is weird for sure, but changing might be involved. probs best suited for another PR. Could create issue?

Reply:
yea fair enough